libsql/internal/ws/driver.go

```diff
@@ -86,18 +86,26 @@ func convertToNamed(args []driver.Value) []driver.NamedValue {
 }
 
 func (s stmt) Exec(args []driver.Value) (driver.Result, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), defaultWSTimeout)
-	defer cancel()
-	return s.c.ExecContext(ctx, s.query, convertToNamed(args))
+	return s.ExecContext(context.Background(), convertToNamed(args))
```

Comment:
Why not `return s.ExecContext(context.Background(), convertToNamed(args))`? `ExecContext` will call `convertToNamed` internally, no?

Reply:
`ExecContext` has a different signature than `Exec`:

```go
func (s stmt) ExecContext(ctx context.Context, args []driver.NamedValue) (driver.Result, error) {}

func (s stmt) Exec(args []driver.Value) (driver.Result, error) {}
```

Since `ExecContext` already gets the `driver.NamedValue`, it doesn't call `convertToNamed` internally.

Reply:
oh - I missed that. Why is ExecContext taking driver.NamedValue not driver.Value? Can't a statement be prepared with positional arguments?

Reply:
ok I see that's the interface https://github.com/golang/go/blob/master/src/database/sql/driver/driver.go#L369